CMakeLists.txt

```diff
@@ -1,28 +1,22 @@
 cmake_minimum_required(VERSION 3.15)
 
-project(learnOpenGL)
+project(
+    learnOpenGL
+    VERSION 0.1.0
+    DESCRIPTION "Small OpenGL projects for study purpose"
+    HOMEPAGE_URL "https://github.com/unsip/learnOpenGL"
+    LANGUAGES CXX
+  )
 
 set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_STANDARD_REQUIRED TRUE)
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra")
```

Comment:
It is a bad practice to modify CMake provided defaults. Use `target_compile_options` instead. In the case of multiple targets, one can make an `INTERFACE` library w/ desired options and link needed targets to it. The proper way is to have your own _custom build type_.